contracts/0.4.24/Lido.sol

```diff
@@ -344,228 +430,96 @@ contract Lido is ILido, StETH, AragonApp {
-
-        if (executionLayerRewardsVaultAddress != address(0)) {
-            executionLayerRewards = ILidoExecutionLayerRewardsVault(executionLayerRewardsVaultAddress).withdrawRewards(
-                (_getTotalPooledEther() * EL_REWARDS_WITHDRAWAL_LIMIT_POSITION.getStorageUint256()) / TOTAL_BASIS_POINTS
-            );
-
-            if (executionLayerRewards != 0) {
-                BUFFERED_ETHER_POSITION.setStorageUint256(_getBufferedEther().add(executionLayerRewards));
-            }
-        }
-
-        // Don’t mint/distribute any protocol fee on the non-profitable Lido oracle report
-        // (when beacon chain balance delta is zero or negative).
-        // See ADR #3 for details: https://research.lido.fi/t/rewards-distribution-after-the-merge-architecture-decision-record/1535
-        if (_beaconBalance > rewardBase) {
-            uint256 rewards = _beaconBalance.sub(rewardBase);
-            distributeFee(rewards.add(executionLayerRewards));
-        }
-    }
-
-    /**
-    * @notice Send funds to recovery Vault. Overrides default AragonApp behaviour
-    * @param _token Token to be sent to recovery vault
-    */
-    function transferToVault(address _token) external {
-        require(allowRecoverability(_token), "RECOVER_DISALLOWED");
-        address vault = getRecoveryVault();
-        require(vault != address(0), "RECOVER_VAULT_ZERO");
-
-        uint256 balance;
-        if (_token == ETH) {
-            balance = _getUnaccountedEther();
-            // Transfer replaced by call to prevent transfer gas amount issue
-            require(vault.call.value(balance)(), "RECOVER_TRANSFER_FAILED");
-        } else {
-            ERC20 token = ERC20(_token);
-            balance = token.staticBalanceOf(this);
-            // safeTransfer comes from overridden default implementation
-            require(token.safeTransfer(vault, balance), "RECOVER_TOKEN_TRANSFER_FAILED");
-        }
-
-        emit RecoverToVault(vault, _token, balance);
-    }
-
-    /**
-    * @notice Returns staking rewards fee rate
-    */
-    function getFee() public view returns (uint16 feeBasisPoints) {
-        return uint16(FEE_POSITION.getStorageUint256());
-    }
-
-    /**
-    * @notice Returns fee distribution proportion
+    * @param _reportTimestamp the moment of the oracle report calculation
+    * @param _timeElapsed seconds elapsed since the previous report calculation
+    * @param _clValidators number of Lido validators on Consensus Layer
+    * @param _clBalance sum of all Lido validators' balances on Consensus Layer
+    * @param _withdrawalVaultBalance withdrawal vault balance on Execution Layer for report block
+    * @param _elRewardsVaultBalance elRewards vault balance on Execution Layer for report block
+    * @param _requestIdToFinalizeUpTo right boundary of requestId range if equals 0, no requests should be finalized
+    * @param _finalizationShareRate share rate that should be used for finalization
```

Comment:
`_finalizationShareRateSimulated` - this is actually simulated value, because math rounding errors inside WQ

Reply:
effective share rate might be differ because of rounding errors inside WQ